src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -25,6 +30,109 @@ public static class Renamer
                 newName, optionSet, cancellationToken);
         }
 
+        /// <summary>
+        /// Applies changes to the given document so that the folders, name, and path from <paramref name="newDocumentInfo"/>
+        /// are used. Will also perform symbol changes to the code in the document, attempting to sync type and namespace
+        /// to the new information.
```

Comment:
it would be nice if we could be a little more explicit about what we're promising here.  i.e. under what circumstances we'll make the symbol change.  i.e. presumably we don't do this if there are multiple top level types in the file (or do we?).